app/src/main/java/com/learner/learndroid/view/RecyclerViewAdapter.java

```diff
@@ -72,37 +73,77 @@ public void onBindViewHolder(@NonNull RecyclerViewHolder recyclerViewHolder, fin
         Log.d(TAG, "Binding view holder for position " + i);
 
         final Item item = productItems.get(i);
+        setViewHolderValues(recyclerViewHolder, item);
+        setViewHolderOnClickListener(recyclerViewHolder, item);
+    }
+
+    /**
+     * Sets the view holder on click listener.
+     *
+     * @param recyclerViewHolder View Holder.
+     * @param item               Item.
+     */
+    private void setViewHolderOnClickListener(@NonNull RecyclerViewHolder recyclerViewHolder, final Item item) {
+        recyclerViewHolder.parentLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
```

Comment:
Shall we try this https://www.phunware.com/blog/android-clean-coding-approach/ ?

Reply:
We can. But I am thinking of doing this the ButterKnife way. So, I think let's try that and see how that goes. If still not good, we'll try this.